mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4465,163 +4506,1149 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
+    g = n = k = c = y = x = ho = wo = hi = wi = 0;
+    for (unsigned i = 0; i < filterLayoutAttr.size(); ++i) {
+      auto filterAttr =
+          filterLayoutAttr.getValue()[i].template cast<StringAttr>();
+      auto inputAttr =
+          inputLayoutAttr.getValue()[i].template cast<StringAttr>();
+      auto outputAttr =
+          outputLayoutAttr.getValue()[i].template cast<StringAttr>();
 
-    int64_t max_lds_align =
-        math::lcm(ABlockCopyDstDataPerWrite_M, BBlockCopyDstDataPerWrite_N,
-                  ThreadGemmAThreadCopySrcDataPerRead_M,
-                  ThreadGemmBThreadCopySrcDataPerRead_N);
+      if (filterAttr.getValue() == "g") {
+        g = filterShape[i];
+      } else if (filterAttr.getValue() == "k") {
+        k = filterShape[i];
+      } else if (filterAttr.getValue() == "c") {
+        c = filterShape[i];
+      } else if (filterAttr.getValue() == "y") {
+        y = filterShape[i];
+      } else if (filterAttr.getValue() == "x") {
+        x = filterShape[i];
+      }
 
-    int64_t KPerBlock =
-        op->getAttr("k_per_block").template cast<IntegerAttr>().getInt();
-    int64_t MPerBlock =
-        op->getAttr("m_per_block").template cast<IntegerAttr>().getInt();
-    int64_t NPerBlock =
-        op->getAttr("n_per_block").template cast<IntegerAttr>().getInt();
+      if (inputAttr.getValue() == "ni") {
+        n = inputShape[i];
+      } else if (inputAttr.getValue() == "hi") {
+        hi = inputShape[i];
+      } else if (inputAttr.getValue() == "wi") {
+        wi = inputShape[i];
+      }
 
-    int64_t AlignedNPerBlock =
-        max_lds_align *
-        math::integer_divide_ceil<int64_t>(NPerBlock, max_lds_align);
+      if (outputAttr.getValue() == "ho") {
+        ho = outputShape[i];
+      } else if (outputAttr.getValue() == "wo") {
+        wo = outputShape[i];
+      }
+    }
 
-    // A matrix in LDS memory, dst of blockwise copy
-    //   be careful of LDS alignment
-    // Original C++ logic:
-    // constexpr auto a_k_m_block_desc = make_native_tensor_descriptor_aligned(
-    //    Sequence<KPerBlock, MPerBlock>{}, Number<max_lds_align>{});
-    // constexpr index_t a_block_space =
-    //    math::integer_least_multiple(a_k_m_block_desc.GetElementSpace(),
-    //    max_lds_align);
-    int64_t AlignedMPerBlock =
-        max_lds_align *
-        math::integer_divide_ceil<int64_t>(MPerBlock, max_lds_align);
-    a_block_space = math::integer_least_multiple(KPerBlock * AlignedMPerBlock,
-                                                 max_lds_align);
+    // compute padding hi/wi.
+    auto hiPadded = hi + leftPadH + rightPadH;
+    auto wiPadded = wi + leftPadW + rightPadW;
+    // calculate gemmKBlocks
+    int64_t gemmN = c * y * x;
+    int64_t gemmM = k;
+    int64_t gemmK = n * ho * wo;
+    int64_t standardBockNum = ((gemmN + 127) / 128) * ((gemmM + 127) / 128);
+    // static const int64_t MaxSubBlockNum = 2048 / standardBockNum;
+    int64_t gemmKBlocks = 1;
```

Comment:
logic for `gemmKBlocks` seem to be duplicated somehow?

Reply:
done